File: php/tests/Sample/Models/TransactionDetail.php
<?php

declare(strict_types=1);

namespace SimpleOrm\Tests\Sample\Models;

use SimpleOrm\Metadata\Attributes\Column;
use SimpleOrm\Metadata\Attributes\ForeignKey;
use SimpleOrm\Metadata\Attributes\Generated;
use SimpleOrm\Metadata\Attributes\Index;
use SimpleOrm\Metadata\Attributes\Key;
use SimpleOrm\Metadata\Attributes\ManyToOne;
use SimpleOrm\Metadata\Attributes\Table;
use SimpleOrm\Metadata\ColumnType;
use SimpleOrm\Types\Decimal;

/** Table `transaction_details`: the many side of `Transaction.details`. `quantity` is an `int32` column — PHP's `int` needs the token override. */
#[Table('transaction_details')]
#[Index(['transactionId'])]
final class TransactionDetail extends BaseModel
{
    #[Key]
    #[Generated]
    #[Column]
    public int $id;

    #[Column]
    #[ForeignKey(Transaction::class)]
    public int $transactionId;

    #[ManyToOne('transactionId')]
    public private(set) ?Transaction $transaction = null;

    #[Column]
    public string $description;

    #[Column(type: ColumnType::Int32)]
    public int $quantity;

    #[Column]
    public Decimal $unitPrice;
}

File: go/orm/sample/transaction_detail.go
package sample

import "github.com/kosmassp/SimpleOrm/go/orm"

// TransactionDetail is table transaction_details (STRICT). Key: id,
// database-generated; transaction_id references transactions. The child side
// of the json_group_array nesting pattern (§7.10).
type TransactionDetail struct {
	BaseModel
	ID            int64 `orm:"column,key,generated"`
	TransactionID int64 `orm:"column"`
	// Transaction is populated only by the library (Level 2 loading); it can never disagree with TransactionID.
	Transaction *Transaction `orm:"many_to_one=TransactionID"`
	Description string       `orm:"column"`
	Quantity    int32        `orm:"column"`
	UnitPrice   orm.Decimal  `orm:"column"`
}

func (TransactionDetail) Entity() orm.EntityDef {
	return orm.EntityDef{
		Source:      orm.Table("transaction_details"),
		ForeignKeys: []orm.ForeignKeyDef{orm.ForeignKey[Transaction]("TransactionID")},
		Indexes:     []orm.IndexDef{orm.Index("TransactionID")},
	}
}
